dusty/config.py

```diff
@@ -61,6 +64,45 @@ def verify_mac_username(username):
     except:
         raise RuntimeError('No user found named {}'.format(username))
 
+def _running_on_mac():
+    return bool(platform.mac_ver()[0])
+
+def _mac_version_is_post_yosemite():
+    version = platform.mac_ver()[0]
+    minor_version = int(version.split('.')[1])
+    return minor_version >= 10
+
+def _load_ssh_auth_post_yosemite():
+    """Starting with Yosemite, launchd was rearchitected and now only one
+    launchd process runs for all users. This allows us to much more easily
+    impersonate a user through launchd and extract the environment
+    variables from their running processes."""
+    user_id = subprocess.check_output(['id', '-u', mac_username])
+    ssh_auth_sock = subprocess.check_output(['launchctl', 'asuser', user_id, 'launchctl', 'getenv', 'SSH_AUTH_SOCK']).rstrip()
+    _set_ssh_auth_sock(ssh_auth_sock)
+
+def _load_ssh_auth_pre_yosemite():
+    """For OS X versions before Yosemite, many launchd processes run simultaneously under
+    different users and different permission models. The simpler `asuser` trick we use
+    in Yosemite doesn't work, since it gets routed to the wrong launchd. We instead need
+    to find the running ssh-agent process and use its PID to navigate ourselves
+    to the correct launchd."""
+    for process in psutil.process_iter():
+        if process.name() == 'ssh-agent':
```

Comment:
No, which is why I designed it to try all the agents it can find and stop if it finds one that gives it a valid listener